src/main/java/com/purbon/kafka/topology/backend/FileBackend.java

```diff
@@ -40,7 +40,7 @@ public void createOrOpen(Mode mode) {
 
   @Override
   public void save(BackendState state) throws IOException {
-    writeLine(state.asJson());
+    writeText(state.asPrettyJson());
```

Comment:
would it be interesting to have this below a flag? I am wondering if the pretty json file would be much bigger? does this makes sense? context: when using terraform (that would be an analogous) json is stored without pretty then you do this pretty stuff with jq when necessary.

Reply:
For one of our test projects, the pretty JSON file is 21% bigger than the compact JSON.
For comparison, the compact JSON file is 49% bigger than the original, text-based format.

So if the goal is both compactness and git-friendlyness, the old format is better. But much harder to maintain in the code.

Reply:
Since the increase in size is not that big I would prefer asPrettyJson. And only add a flag later if there is actual demand.

Reply:
fair for me, sorry I had to ask ;-)